src/client.ts

```diff
@@ -643,19 +646,21 @@ export class Client extends EventEmitter {
     }
 
     if (this.hardReconnectRequired) {
-      let jidPromise: Promise<any>;
-      if (this.config.jid) {
-        jidPromise = Promise.resolve(this.config.jid);
-      } else {
-        const jidRequestOpts: RequestApiOptions = {
-          method: 'get',
-          host: this.config.apiHost,
-          authToken: this.config.authToken,
-          logger: this.logger,
-          customHeaders: this.config.customHeaders
-        };
-        jidPromise = this.http.requestApi('users/me', jidRequestOpts)
-          .then(res => res.data.chat.jabberId);
+      // Use stored JID if we have one, otherwise use the provided JID or grab one.
+      if (!this.initialJid) {
+        if (this.config.jid) {
```

Comment:
I'm not sure if this is actually needed, it may be redundant because we end up assigning `this.config.jid = this.initialJid` later too.

Reply:
I think the behavior is mostly the same as it was before.

The idea was - if we don't have a saved JID, then we would try to use the one in the config, but if we don't have one there then we'd fetch one.

And if we fetched one we needed to assign it to the config/saved jid. I'm not sure there's _harm_ in it as it is, but its definitely a bit confusing.

Reply:
Yeah I'm following the intent here and I think you're right. I don't think there are any issues as is but it is strange.  I'll still approve it because I think we should still be fine but out of curiosity, would there be any benefit to `this.config.jid = this.initialJid` being moved into the else block here?  Spitballing more than anything and definitely not saying it's a necessary change, just curious more than anything.